File: case1.py
weights = '56 65 74 100 99 68 86 180 90'

def sumOfDigit(n):  
    sum = 0
    while (n > 0):
        sum += n % 10
        n = n // 10 
    return sum

def takeFirst(elem):
    return elem[0]

def sortByWeight (string):
    if weights:
        weigths_hashmap = [] #sum_weigth : weigth
        weights_string = ''
        weights_array = string.split(' ')
        for weight in weights_array:
            weigths_hashmap.append((sumOfDigit(int(weight)),weight))
        weigths_hashmap.sort(key = takeFirst)
        for element in weigths_hashmap:
            weights_string += element[1] + ' '
        return weights_string
    else:
        return 'no weigths submitted'

print(sortByWeight(weights))

File: case1.js

let weights = "56 65 74 100 99 68 86 180 90"

function sumOfDigit(n){
    let addition = 0;
    while (n != 0){
        addition += n % 10
        n = Math.floor(n / 10) 
    }
    return addition
}  

function takeSecondElem (element){
    return element[1]
}

function orderWeight (string){
    let weight_array = string.split(' ');
    let weight_hash = []
    let weight_resut=[];
    weight_array.forEach(element => {
        weight_hash.push([sumOfDigit(element),element])
    });
    weight_hash.sort(function(a,b){
        return a == b 
        ? a[1].localeCompare(b[1])
        : (a[0] - b[0]);
    })
    weight_hash.forEach(weigth =>{
        weight_resut.push(takeSecondElem(weigth))
    })
    return weight_resut.join(' ')
}

console.log(orderWeight(weights));